### PR TITLE
GH Actions: Nightly builds improvements

### PR DESCRIPTION
**Pull request by marcospereira**
## Changes

This changes some of the existing jobs in a relevant way:

### Before

| Workflow                            | JDK | Scala            | `akka.genjavadoc.enabled` | `akka.test.multi-in-test` | `akka.cluster.assert` |
|:------------------------------------|:----|:-----------------|:--------------------------|:--------------------------|:----------------------|
| `nightly-cluster-metrics-sigar.yml` | 8   | default (`2.13`) | default (`false`)         | default (`true`)          | `on`                  |
| `nightly-jdk8.yml`                  | 8   | default (`2.13`) | default (`false`)         | default (`true`)          | `on`                  |
| `nightly-jdk8-scala213.yml`         | 8   | `2.13`           | default (`false`)         | default (`true`)          | `on`                  |
| `nightly-jdk11.yml`                 | 11  | default (`2.13`) | `true`                    | `false`                   | default (`off`)       |

So, we can see `nightly-jdk8.yml` and `nightly-jdk8-scala213.yml` are actually duplicated (since Scala 2.13 is the default).

### After

We are introducing only one new combination, and removing the duplication of JDK 8/Scala 2.13:

| Job                             | JDK | Scala            | `akka.genjavadoc.enabled`  | `akka.test.multi-in-test` | `akka.cluster.assert` |
|:--------------------------------|:----|:-----------------|:---------------------------|:--------------------------|:----------------------|
| `nightly-cluster-metrics-sigar` | 8   | default (`2.13`) | do not apply*              | default (`true`)          | `on`                  |
| `jdk-nightly-tests`             | 8   | `2.12`           | do not apply*              | `false`                   | `on`                  |
| `jdk-nightly-tests`             | 8   | `2.13`           | do not apply*              | `true`                    | `on`                  |
| `jdk-nightly-tests`             | 11  | `2.12`           | `true`                     | `false`                   | `on`                  |
| `jdk-nightly-tests`             | 11  | `2.13`           | `true`                     | `true`                    | `on`                  |

*`doc` step running only for JDK 11.

The new combination is JDK 11 + Scala 2.12. But there are other changes too:

- JDK 11 combination now runs with `akka.cluster.assert=on`
- JDK 11 + Scala 2.13 now runs with `akka.test.multi-in-test=true`

---

This new setup seems more consistent and easier to maintain in my option. It is also pretty straightforward to change the matrix later and have more handpicked combinations instead of an `n` x `m` matrix with some conditionals (see an example on [Github Action docs](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-environment-variables-in-a-matrix)).

### CODE REVIEW

File: .github/workflows/nightly-builds.yml
Comment (patriknw) [inline .github/workflows/nightly-builds.yml:54]: I guess this is ok for 2x2, but not sure we have to run all combinations when we expand this to let's say 3x3

**Review by patriknw — COMMENTED** [review]
looking good, I guess this is replacing the existing nightly-* workflows and those should be removed?

File: .github/workflows/nightly-builds.yml
Comment (marcospereira) [reply .github/workflows/nightly-builds.yml:61]: Yes, I agree. The description is up to date to compare the existing jobs vs. what is proposed by this PR.

File: .github/workflows/nightly-builds.yml
Comment (patriknw) [inline .github/workflows/nightly-builds.yml:95]: We have tried to make tests handle rather random pauses of 2-3 seconds on Jenkins. Default timeout for asserts is 3 seconds. Multiplying that with 10 makes no sense to me. Pauses of 30 seconds from the CI environment is not something we can deal with, and I don't gh actions will have such delays.

I can understand if we want to try with `timefactor=2` but if things are still flaky with that we have other hardening work to do.

**Review by patriknw — COMMENTED** [review]
looking good, aside from the timefactor

**Review by marcospereira — COMMENTED** [review]
Some comments.

File: .github/workflows/nightly-builds.yml
Comment (marcospereira) [reply .github/workflows/nightly-builds.yml:95]: > I can understand if we want to try with `timefactor=2` but if things are still flaky with that we have other hardening work to do.

We can go with 2, I agree. I was using the same value we use now for Lagom.

Comment (marcospereira) [reply .github/workflows/nightly-builds.yml]: Done in 35dbab7b582d26d4d4cb36675bb1d93186bd5f03.

File: project/AkkaBuild.scala
Comment (patriknw) [reply project/AkkaBuild.scala:207]: It has been very useful for correlating test failures (timeouts) with GC pauses. Especially now with a new environment we should have it enabled.

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
@marcospereira Created issue https://github.com/akka/akka/issues/30559 for the test failure, but there is also something about the email if you would like to look before merging.

**Comment by marcospereira** [comment]
> but there is also something about the email if you would like to look before merging.

Yeah, that is because the secrets aren't available for pull requests.  This should be good to go now after https://github.com/akka/akka/pull/30551/commits/81afa5bc35a8edd3ca0ee1f5dde3fdae9a362861.

File: .github/workflows/nightly-builds.yml
Comment (johanandren) [reply .github/workflows/nightly-builds.yml:102]: Create an issue to track (so we don't forget)?

File: .github/workflows/nightly-builds.yml
Comment (johanandren) [inline .github/workflows/nightly-builds.yml:124]: This in a matrix - does it mean we could publish a snapshot for Scala 2.13 while 2.12 failed? (Not entirely sure if that is a problem, just something I thought about)

File: project/AkkaBuild.scala
Comment (johanandren) [reply project/AkkaBuild.scala:24]: I thought it was used in more places but a quick search says only for the GC-logs, so we could probably remove it.

**Review by johanandren — APPROVED** [review]
LGTM, with a few questions, good work @marcospereira !

File: .github/workflows/nightly-builds.yml
Comment (marcospereira) [reply .github/workflows/nightly-builds.yml:102]: Done: https://github.com/akka/akka/issues/30568

File: .github/workflows/nightly-builds.yml
Comment (marcospereira) [reply .github/workflows/nightly-builds.yml:124]: This is publishing local, so we won't have a problem with inconsistent snapshots unless I'm missing something from what you said.

File: project/AkkaBuild.scala
Comment (marcospereira) [reply project/AkkaBuild.scala:24]: I've declared it because the logic now considers both `akka.ci-server` property and `CI` env var, and I thought that having a `CliOption` makes it slightly clearer. But yeah, I can remove it and check both directly.

**Comment by marcospereira** [comment]
@johanandren got this merged so that we can have these running for nightly builds already. I can follow up on your questions later in future pull requests. :-)

**Pull request merged**
